### ISSUE TITLE
Frontlight dialog lags if activated through a gesture

### ISSUE THREAD

**Issue by Fractalii** [koreader/koreader#12872]
* KOReader version: 2024.11
* Device: Kindle Paperwhite 2021

#### Issue

If you use a gesture to open frontlight menu (probably others as well) in 2024.11 update, it will partially render (and lag for a second), then happens page refresh, and only after the menu appears. In the previous 2024.07 update the menu appeared at once without any problems.

#### Steps to reproduce

Set 'Show frontlight dialog' to a gesture (tap corner in my case) and use it

##### `crash.log`

<details>
  <summary>crash.log</summary>
12/08/24-12:20:50 INFO  Restoring user input handling 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 57 (ABS_MT_TRACKING_ID), value: 0, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 53 (ABS_MT_POSITION_X), value: 66, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 54 (ABS_MT_POSITION_Y), value: 22, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 58 (ABS_MT_PRESSURE), value: 66, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 48 (ABS_MT_TOUCH_MAJOR), value: 80, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1733649951.648727 
12/08/24-12:20:51 DEBUG GestureDetector:probeClockSource: Touch event timestamps appear to use CLOCK_REALTIME 
12/08/24-12:20:51 DEBUG slot 0 in tap state... 
12/08/24-12:20:51 DEBUG set up hold timer 
12/08/24-12:20:51 DEBUG AutoSuspend: onInputEvent 
12/08/24-12:20:51 DEBUG input event => type: 3 (EV_ABS), code: 57 (ABS_MT_TRACKING_ID), value: -1, time: 1733649951.698562 
12/08/24-12:20:51 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1733649951.698562 
12/08/24-12:20:51 DEBUG slot 0 in tap state... 
12/08/24-12:20:51 DEBUG Contact:handleDoubleTap: single tap detected @ 66 22 
12/08/24-12:20:51 DEBUG AutoSuspend: onInputEvent 
12/08/24-12:20:51 DEBUG ReaderLink:getLinkFromPosition @ 66 22 from a_xpointer: nil to link_xpointer:  
12/08/24-12:20:51 DEBUG FocusManager: Move focus position to: 1 , 1 
12/08/24-12:20:51 DEBUG setDirty via a func from widget FrontLightWidget 
12/08/24-12:20:51 DEBUG ImageWidget: _render'ing resources/icons/mdlight/close.svg 50 50 
12/08/24-12:20:51 DEBUG renderSVG resources/icons/mdlight/close.svg 2.0833333333333 24 24 > 50 50 0 0 
12/08/24-12:20:51 DEBUG cache image|resources/icons/mdlight/close.svg|50|50|flat 
12/08/24-12:20:51 DEBUG ImageWidget: initial offsets 0 0 
12/08/24-12:20:51 DEBUG show widget: FrontLightWidget 
12/08/24-12:20:51 DEBUG setDirty nil from widget FrontLightWidget w/ NO region; dithering: nil 
12/08/24-12:20:51 DEBUG setDirty via a func from widget FrontLightWidget 
12/08/24-12:20:51 DEBUG painting widget: FrontLightWidget 
12/08/24-12:20:51 DEBUG Found font: NotoSansCJKsc-Regular.otf in ./fonts/noto/NotoSansCJKsc-Regular.otf 
12/08/24-12:20:51 DEBUG blitFrom 1132 357 0 0 50 50 
12/08/24-12:20:51 DEBUG _refresh: Enqueued ui update for region 52 1094 1131 92 dithering: false 
12/08/24-12:20:51 DEBUG _refresh: Enqueued ui update for region 27 330 1182 988 dithering: false 
12/08/24-12:20:51 DEBUG triggering refresh {
  mode = "ui",
  region = "1182x988+27+330"
} --[[table: 0xb5811d00]] 
12/08/24-12:20:51 DEBUG refresh: ui-mode 27 330 1182 988 nil 
12/08/24-12:20:51 DEBUG mxc_update: 1182x988 region @ (27, 330) with marker 4 (WFM: 257 & UPD: 0) 
12/08/24-12:20:51 INFO  Inhibiting user input 
12/08/24-12:20:51 DEBUG _refresh: Enqueued partial update for region 0 0 1236 1648 dithering: false 
12/08/24-12:20:51 DEBUG setDirty partial from widget ReaderUI w/ NO region; dithering: nil 
12/08/24-12:20:51 DEBUG painting widget: ReaderUI 
12/08/24-12:20:51 DEBUG readerview painting "1236x1648+0+0" to 0 0 
12/08/24-12:20:51 DEBUG CreDocument: goto page 305 flow 0 
12/08/24-12:20:51 DEBUG painting widget: FrontLightWidget 
12/08/24-12:20:51 DEBUG blitFrom 1132 357 0 0 50 50 
12/08/24-12:20:51 DEBUG triggering refresh {
  mode = "partial",
  region = "1236x1648+0+0"
} --[[table: 0xb577d350]] 
12/08/24-12:20:51 DEBUG refresh: partial 0 0 1236 1648 nil 
12/08/24-12:20:51 DEBUG refresh: wait for completion of (previous) marker 4 
12/08/24-12:20:52 DEBUG mxc_update: 1236x1648 region @ (0, 0) with marker 5 (WFM: 4 & UPD: 1) 
12/08/24-12:20:52 DEBUG refresh: wait for completion of marker 5 
12/08/24-12:20:52 INFO  Restoring user input handling 
</details>

**Comment by NiLuJe** [comment]
Might be the thing I was mentioning in https://github.com/koreader/koreader/issues/12867#issuecomment-2525297983 ;).

(i.e., t also applies to MTK on Kobo).

**Comment by shamanNS** [comment]
Yeah, I also use a gesture (tap top left corner of the screen) to show/hide frontlight widget and noticed the same issue. Also recently noticed that the dialog window that offers to open a book for reading once transfer via CloudStorage (using FTP server) finishes also has problems and often doesn't render "cancel" and "ok" buttons at the bottom of said window.

**Comment by mergen3107** [comment]
Can reproduce on PW6 and Scribe on 2024.11-224. 

Frontlight widget shows up the same as from the top menu

**Comment by shamanNS** [comment]
> Sorry, I meant **cannot reproduce** :D
> 
> [@shamanNS](https://github.com/shamanNS) Can you reproduce it on current nightly?

@mergen3107 
I don't use nightlies 😁 (I"m not on armhf firmware)

**Comment by Fractalii** [comment]
The problem is present on 2025.08
